src/test/unit/controls/Graph/GraphAxes-spec.js

```diff
@@ -685,4 +695,473 @@ describe("Graph - Axes", () => {
+                    .textContent
+            ).toBe(
+                "Project long display value which is only for testing, Project long display value which is only for testing Project long display value which is only for testing, Project long d..."
+            );
+            expect(
+                fetchElementByClass(styles.axisLabelY2).querySelector("text")
+                    .textContent
+            ).toBe(
+                "Project long display value which is only for testing, Project long display value which is only for testing Project long display value which is only for testing, Project long d..."
+            );
+            done();
+        });
+        it("Does not truncate if not long", (done) => {
+            graph = new Graph(getAxes(axisDefault));
+            expect(
+                fetchElementByClass(styles.axisLabelX).querySelector("text")
+                    .textContent
+            ).toBe("Some X Label");
+            expect(
+                fetchElementByClass(styles.axisLabelY).querySelector("text")
+                    .textContent
+            ).toBe("Some Y Label");
+            expect(
+                fetchElementByClass(styles.axisLabelY2).querySelector("text")
+                    .textContent
+            ).toBe("Some Y2 Label");
+            done();
+        });
+    });
+    describe("Adds clickHandler for Label", () => {
+        beforeEach(() => {
+            graph.destroy();
+            axisDefault.y2 = {
+                show: true,
+                label: "Some Y2 Label",
+                lowerLimit: 0,
+                upperLimit: 20
+            };
+        });
+        afterEach(() => {
+            axisDefault.y2 = "";
+        });
+        it("Click functionality disabled when small labels are provided and onLabelClick is not provided", (done) => {
+            const onClickPrimaryFunctionSpy = sinon.spy();
+            graph = new Graph(getAxes(axisDefault));
+            triggerEvent(
+                fetchElementByClass(`${styles.axisLabelX} text`),
+                "click",
+                () => {
+                    expect(onClickPrimaryFunctionSpy.calledOnce).toBeFalsy();
+                    done();
+                }
+            );
+            triggerEvent(
+                fetchElementByClass(`${styles.axisLabelY} text`),
+                "click",
+                () => {
+                    expect(onClickPrimaryFunctionSpy.calledOnce).toBeFalsy();
+                    done();
+                }
+            );
+            triggerEvent(
+                fetchElementByClass(`${styles.axisLabelY2} text`),
+                "click",
+                () => {
+                    expect(onClickPrimaryFunctionSpy.calledOnce).toBeFalsy();
+                    done();
+                }
+            );
+        });
+        it("Click functionality disabled when, small label and onLabelClick function is provided", (done) => {
```

Comment:
Did we cover the scenario?

- If the small label is provided and `onLabelClick` function is **not** provided

Reply:
@Veera81 that case got covered already. please take a look at this [code](https://github.com/cerner/carbon-graphs/pull/242/commits/3d75d6b1109fb3da7bad04f51b1228bd6bcaaf93#diff-7fb4a16b40433875e7e73811189b4f84R164-R178) .